compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -758,7 +758,8 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
           MissingType(tycon.prefix, tycon.name).toMessage.message
         case _ =>
           i"Cannot resolve reference to $tp"
-      throw FatalError(msg)
+      if ctx.isBestEffort then report.error(msg)
+      else throw FatalError(msg)
```

Comment:
This one is a bit worrying. The doc comment explicitly says that it won't let `!tp1.exists` pass through. Presumably, callers rely on that not to crash just after.

Reply:
I have added a comment to the doc of the method to reflect this. In the situations I tested, this was (luckily) not a problem, and the only solution here would be to keep throwing FatalError, which would stop the best-effort compilation (with a crash, no less). What I'm trying to say here is, by not throwing this FatalError we minimize crashes, and maximize successful best effort compilations